pytorch_pfn_extras/config_types.py

```diff
@@ -1,9 +1,14 @@
 import warnings
+from typing import Any, Callable, Dict, Optional, TYPE_CHECKING
 
-from .config import Config
+from pytorch_pfn_extras import config
 
 
-def optuna_types(trial):
+if TYPE_CHECKING:
+    import optuna
+
+
+def optuna_types(trial: 'optuna.Trial'):
```

Comment:
The public API seems `optuna.trial.Trial`.
https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html#optuna.trial.Trial